src/pages/events/createEvents/CreateEventFirstPage.tsx

```diff
@@ -0,0 +1,412 @@
+
+const CreateEventFirstPage = () => {
+  const classes = useCreateEventStyles()
+  // const navigate = useNavigate()
+  const UseEventContext = () => useContext(EventContext)
+  const {dispatch} = UseEventContext()
+
+  const [evaluatorsCount, setEvaluatorsCount] = useState<number>(0)
+  const [evaluateesCount, setEvaluateesCount] = useState<number>(0)
+  const [eventTitle, setEventTitle] = useState<string>('')
+  const [evaluatorsList, setEvaluatorsList] = useState<IEvaluator[]>([])
+  const [evaluateesList, setEvaluateesList] = useState<IEvaluatee[]>([])
+  const [isTopManagerChecked, setIsTopManagerChecked] = useState<boolean>(true)
+  const [foundTopManager, setFoundTopManager] = useState<boolean>(false)
+  const [topManagerId, setTopManagerId] = useState<string>('')
+  const [evaluatorCountError, setEvaluatorCountError] = useState<boolean>(false)
+  const [evaluateeCountError, setEvaluateeCountError] = useState<boolean>(false)
+
+  const [topManagerEmptyObjInfo] = useState<ITopManagerEmptyObjInfoType>({
+    id: uuidv4(),
+    header: 'Evaluator 1 (Top Manager)',
+    name: 'Norvik Abdalian',
+    position: 'CTO',
+    email: 'norvik@adrack.com',
+  })
+
+  const [evaluatorObjInfo] = useState<IEvaluator>({
+    id: uuidv4(),
+    header: 'Evaluator 1',
+    name: 'Name / Surname',
+    nameValue: '',
+    position: 'Position',
+    positionValue: '',
+    email: 'Email',
+    emailValue: '',
+  })
+
+  const [evaluateeObjInfo] = useState<IEvaluatee>({
+    id: uuidv4(),
+    header: '',
+    name: 'Name / Surname',
+    nameValue: '',
+    position: 'Position',
+    positionValue: '',
+    date: 'Hire date',
+    dateValue: undefined,
+    salary: 'Monthly Salary',
+    salaryValue: '',
+  })
+
+  const navigateToCriteriasAndSendDataToContext = () => {
+    dispatch({type: createEventReducerTypes.eventTitle, eventTitle})
+    dispatch({type: createEventReducerTypes.evaluators, evaluators: evaluatorsList})
+    dispatch({type: createEventReducerTypes.evaluatees, evaluatees: evaluateesList})
+    dispatch({
+      type: createEventReducerTypes.updateActivePageToCriterias,
+      activePage: 'criteriasPage',
+    })
+  }
+
+  const label = {inputProps: {'aria-label': 'Checkbox demo'}}
+
+  const handleChangeCheckbox = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setIsTopManagerChecked(e.target.checked)
+  }
+
+  const onChangeEvaluatorData = (id: string, value: string, changingValue: string) => {
+    const newArray = evaluatorsList.map((item: IEvaluator) => {
+      if (item.id === id) {
+        return {...item, [changingValue]: value}
+      }
+      return item
+    })
+    setEvaluatorsList(newArray)
+  }
+
+  const onChangeEvaluateeData = (id: string, value: string, changingValue: string) => {
+    const newArray = evaluateesList.map((item: IEvaluatee) => {
+      if (item.id === id) {
+        return {...item, [changingValue]: value}
+      }
+      return item
+    })
+    setEvaluateesList(newArray)
+  }
```

Comment:
almost the same function

Reply:
They are different fields, and two functions are needed